test/opengl/user_state_check.cpp

```diff
@@ -0,0 +1,82 @@
+#include <csetjmp>
+#include <unistd.h>
+
+#if defined(__APPLE__)
```

Comment:
rewrap_texture in master now starts with:

// Test doesn't build on windows, because OpenGL on windows is a nightmare.
#ifdef _WIN32
#include <stdio.h>
int main() {
    printf("Skipping test on Windows\n");
    return 0;
}
#else

#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>
#include <stddef.h>
#include "../src/runtime/mini_opengl.h"

#include "Halide.h"
#include "HalideRuntimeOpenGL.h"

using namespace Halide;

extern "C" void glGenTextures(GLsizei, GLuint *);
extern "C" void glTexParameteri(GLenum, GLenum, GLint);
extern "C" void glBindTexture(GLenum, GLuint);
extern "C" void glTexImage2D(GLenum, GLint, GLint, GLsizei, GLsizei, GLint,
GLenum, GLenum, const GLvoid *);

It's possible we can switch back to that header block again - I extern
declared everything because I was having trouble on windows, but then I
just gave up on windows and disabled the tests that need to actually link
to libGL and expect the symbols to be there.

On Mon, Aug 8, 2016 at 6:15 AM, ronen barzel <notifications@github.com>
wrote:

> In test/opengl/user_state_check.cpp
> <https://github.com/halide/Halide/pull/1381#discussion_r73872392>:
>
> > @@ -0,0 +1,82 @@
> > +#include <csetjmp>
> > +#include <unistd.h>
> > +
> > +#if defined(__APPLE__)
>
> @abadams <https://github.com/abadams> I'm confused, user_state_check.cpp
> and rewrap_texture.cpp both have the same boilerplate for including GL
> headers:
>
> #if defined(__APPLE__)
> #include <OpenGL/gl.h>
> #else
> #include <GL/gl.h>
> #endif
>
> Is there something else I'm not seeing?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/halide/Halide/pull/1381/files/781151b93e49b3d447e6ef4474af1a00244cc58e#r73872392>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAfdRmECx0cLjI2GsJyZagxXoAh4kyl3ks5qdyvigaJpZM4JNh1R>
> .
>